R/rowClust.R

```diff
@@ -0,0 +1,112 @@
+#' Clusters rowData using \code{fastcluster::hclust}
+#'
+#' @details
+#' \code{rowClust} Performs hierarchical clustering on rowData of the 
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}, 
+#' by using the dissimilarity function provided at \code{dissimilarity_FUN}.
+#'
+#' @param x a
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object.
+#'   
+#' @param abund_values name of the assay containing the data.
+#' 
+#' @param name name of the new column at rowData where to place cluster ids.
+#' 
+#' @param clustering_method clustering method to be used, same as methods 
+#' available at \code{\link[hclust]{hclust}}.
+#' 
+#' @param transposed whether the data should be transposed.
+#' 
+#' @param dissimilarity_FUN function to compute the dissimilarity matrix
+#' 
+#' @param k number of clusters (same as in \code{\link[hclust]{hclust}}).
+#' 
+#' @param h cut-off height (same as \code{\link[hclust]{hclust}})
+#'
+#' @return \code{x} with additional \code{\link{rowData}} named \code{*name*}
+#'
+#' 
+#' @name rowClust
+#' 
+#' @export
+#'
+#' @examples
+#' # Load example data
+#' library(mia)
+#' data("GlobalPatterns")
+#' tse <- GlobalPatterns
+#' # Agglomerate to Class rank
+#' tse <- agglomerateByRank(tse, rank="Class")
+#' # Cluster using spearman correlation as a dissimilarity and ward.D2 
+#' # for clustering; with results having 6 clusters
+#' tse <- rowClust(tse, k = 6)
+#' # Using cosine dissimilarity, average clustering and 0.6 as the height
+#' # of cut-off
+#' tse <- rowClust(tse, 
+#'     dissimilarity_FUN=function(x) 1 - coop::cosine(x),
```

Comment:
Is this our recommended method? Which distance methods other papers have used? We can give interesting usage examples like this but we should be clear on what we think is a good default choice.

Reply:
https://www.nature.com/articles/nature11319 has used `Hierarchical Ward-linkage clustering based on the
Spearman correlation`.
and the second https://pubmed.ncbi.nlm.nih.gov/31370880/ `Approximate Nearest Neighbor heuristic for near-exhaustive average linkage clustering`, however very big dateset computed on a cloud.

Reply:
The first one OK for now